### PR TITLE
Prefs [TASK][26313] Handle Events for new message screen

### PR DESCRIPTION
**Pull request by toandv98**
## Related Tickets

- [#Task 26313](https://edu-redmine.sun-asterisk.vn/issues/26313)

## WHAT

- ###  Handle Events for new message screen

## HOW

- Click plan or report button
- Click send button
- Send message using API
- Save message to SQLite

## Review Checklist

| Category      | View Point                                                   | Description                                                  | Expected Reviewer Answer | Reviewer1 (ToanDV) | Reviewer2 (name)   |
| ------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ | ------------------ | ------------------ |
| General       | Code should be self-explanatory.                             | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Is this function or class too big? If yes, is the function or class having too many responsibilities? |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| General       | Does the code follow DRY?: Do not Repeat Yourself            | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Exception handling and cleanup (dispose) resources           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Does the code build and execute correctly?                   |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the source code have end-line errors?                   |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| Conventions   | No hard coding, use constants/configuration values           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Redmine       | Does the ticket follow Sun* Redmine working process?         | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |

## Notes (Optional)

*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/edu/sun/oereminder/data/model/Message.kt [hunk 1]
```diff
@@ -71,5 +74,14 @@ data class Message(
         const val FIRST_LAST_END_LINE = "^\\n|\\n\$"
         const val HTML_END_LINE = "<br>"
         const val END_LINE = "\n"
+        private const val PLAN_TITLE_REGEX = "<font color='#000'><b>1. Plan:</b></font><br>"
```
Comment (tuanpa-0908) [inline app/src/main/java/com/edu/sun/oereminder/data/model/Message.kt]: t recommend để trong resource cho dễ quản lý nhé

File: app/src/main/java/com/edu/sun/oereminder/ui/message/MessageDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            textLabelReport.isVisible = !isCheck
+            layoutActual.isVisible = isCheck
+            layoutNext.isVisible = isCheck
+            layoutIssue.isVisible = isCheck
+        }
+        switchReport.isChecked = !isPlan
+    }
+
+    private fun setResult(): Boolean {
+        val code = when {
```
Comment (tuanpa-0908) [inline app/src/main/java/com/edu/sun/oereminder/ui/message/MessageDialogFragment.kt:150]: Dùng if else đi ông. Có 2 trường hợp

File: app/src/main/java/com/edu/sun/oereminder/ui/message/MessageDialogFragment.kt [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        }
+        setFragmentResult(
+            REQUEST_SEND,
+            bundleOf(REQUEST_SEND to code, KEY_MESSAGE to getMessageBody())
+        )
+        dismiss()
+        return true
+    }
+
+    private fun getMessageBody() = if (isPlan) getString(
```
Comment (tuanpa-0908) [inline app/src/main/java/com/edu/sun/oereminder/ui/message/MessageDialogFragment.kt]:
```suggestion
    private fun getMessageBody() = if (isPlan) {
        getString(
```

File: app/src/main/java/com/edu/sun/oereminder/ui/message/MessageDialogFragment.kt [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        R.string.plan_format,
+        edtMessage.text,
+        if (isEdit) {
+            message?.sendCalendar?.toDate(REPORT_DATE_FORMAT)
+        } else {
+            GregorianCalendar().toDate(REPORT_DATE_FORMAT)
+        },
+        edtPlan.text
+    ) else getString(
```
Comment (tuanpa-0908) [inline app/src/main/java/com/edu/sun/oereminder/ui/message/MessageDialogFragment.kt]:
```suggestion
    ) else {
        getString(
```

File: app/src/main/java/com/edu/sun/oereminder/ui/message/MessageDialogFragment.kt [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        R.string.report_format,
+        edtMessage.text,
+        edtPlan.text,
+        edtActual.text,
+        edtNext.text,
+        edtIssue.text
+    )
+
+    private fun String.removeHtmlTag() =
```
Comment (tuanpa-0908) [inline app/src/main/java/com/edu/sun/oereminder/ui/message/MessageDialogFragment.kt]: Cho ra String+.kt

File: app/src/main/java/com/edu/sun/oereminder/ui/report/ReportPresenter.kt [hunk 6]
```diff
@@ -35,11 +40,75 @@ class ReportPresenter(private val messageRepository: MessageRepository) :
...
+            if (1000 * sendTime > firstMillisOfDay()) {
+                this@ReportPresenter.isPlan = isReport
+                view?.showFab(!isReport, isReport)
+            } else {
+                view?.showFab(true)
+            }
+        }
+    }
+
+    private fun newReport(messageBody: String) {
```
Comment (tuanpa-0908) [inline app/src/main/java/com/edu/sun/oereminder/ui/report/ReportPresenter.kt]:
```suggestion
    private fun addNewReport(messageBody: String) {
```

Comment (tuanpa-0908) [reply app/src/main/java/com/edu/sun/oereminder/ui/report/ReportPresenter.kt]: Tên function là động từ

**Comment by tuanpa-0908** [comment]
Rebase đi nhé. 2 commit

File: app/src/main/java/com/edu/sun/oereminder/ui/message/MessageDialogFragment.kt (hunk 2 above)
Comment (toandv98) [reply app/src/main/java/com/edu/sun/oereminder/ui/message/MessageDialogFragment.kt:150]: T đang thêm chức năng xóa report, nên nó sẽ là 2 case :))

**Review by aug16vcc — APPROVED** [review]

**Review by tuanpa-0908 — APPROVED** [review]

**Pull request merged**